awswrangler/s3/_read_parquet.py

```diff
@@ -969,112 +732,96 @@ def read_parquet_table(
     df = read_parquet(
         path=paths,
         path_root=path_root,
+        dataset=True,
         path_suffix=filename_suffix if path_root is None else None,
         path_ignore_suffix=filename_ignore_suffix if path_root is None else None,
         columns=columns,
         validate_schema=validate_schema,
-        categories=categories,
-        safe=safe,
-        map_types=map_types,
+        coerce_int96_timestamp_unit=coerce_int96_timestamp_unit,
         chunked=chunked,
-        dataset=True,
         use_threads=use_threads,
+        parallelism=parallelism,
         boto3_session=boto3_session,
-        s3_additional_kwargs=s3_additional_kwargs,
+        arrow_additional_kwargs=arrow_additional_kwargs,
     )
+
     partial_cast_function = functools.partial(
         _data_types.cast_pandas_with_athena_types, dtype=_extract_partitions_dtypes_from_table_details(response=res)
     )
-
     if isinstance(df, pd.DataFrame):
         return partial_cast_function(df)
-
     # df is a generator, so map is needed for casting dtypes
     return map(partial_cast_function, df)
```

Comment:
I am worried about these transformations too. Most likely to be too slow on a large DataFrame

Reply:
Should we do an `apply` in distributed scenario?